yapapi/strategy.py

```diff
@@ -25,6 +27,28 @@
 class MarketStrategy(DemandDecorator, abc.ABC):
     """Abstract market strategy."""
 
+    valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+
+    def set_valid_prop_value_ranges(
+        self, valid_prop_value_ranges: Dict[str, Tuple[Optional[float], Optional[float]]]
+    ) -> None:
+        self.valid_prop_value_ranges = valid_prop_value_ranges
+
+    async def answer_to_provider_offer(
+        self, our_demand: DemandBuilder, provider_offer: rest.market.OfferProposal
+    ) -> DemandBuilder:
+        updated_demand = deepcopy(our_demand)
```

Comment:
Alternative: this function can modify `our_demand` if all properties have valid values (instead of returning a new `DemandBuilder` object).